### PR TITLE
[IMP] *_sale_*: move overridable configurator logic from controllers to models

### LINKED ISSUE #179816: [master] sale: can't extend combo configurator controller
### Odoo Version

- [ ] 15.0
- [ ] 16.0
- [ ] 17.0
- [X] Other (specify)

### Steps to Reproduce

- Import controller in file
`from odoo.addons.sale.controllers.combo_configurator import SaleComboConfiguratorController`
- Create Class inheriting original controller
`Class ComboConfigExt(SaleComboConfiguratorController):`
- Overwrite a function and add logger
```
@route()
def sale_combo_configurator_get_data(
    self,
    product_tmpl_id,
    quantity,
    date,
    currency_id=None,
    company_id=None,
    pricelist_id=None,
    selected_combo_items=None,
):
    _logger.info('test')
    return super().sale_combo_configurator_get_data(**kwargs)
```

**Outcome:**
- Logger doesn't output and it just uses the base function

**Workaround:**
The only solution to this that I have found is extending from website_sale WebsiteSaleComboConfiguratorController but would then require the custom module to have a dependency on website_sale which isn't necessary.

### Log Output

_No response_

### Support Ticket

_No response_

### PR DESCRIPTION
**Pull request by ltinel**
This allows to rely on the framework's inheritance instead of python's inheritance. Indeed, there was an issue where `sale_subscription` overrides were ignored by the framework. This change applies to both the product and combo configurators (and extracts any duplicated logic into shared methods).

This change also fixes the strikethrough price in combo and product configurators.

task-4263961

Enterprise PR: https://github.com/odoo/enterprise/pull/70533
Closes https://github.com/odoo/odoo/issues/179816

### CODE REVIEW

File: addons/website_sale/controllers/main.py [hunk 1]
```diff
@@ -2234,6 +2234,6 @@ def _apply_taxes_to_price(price, product_or_template, currency):
             fiscal_position = request.website.fiscal_position_id.sudo()
```
Comment (Feyensv) [inline addons/website_sale/controllers/main.py:2234]: `_apply_taxes_to_price` can be moved to configurator controller as we discussed

File: addons/website_sale/models/product_template.py [hunk 2]
```diff
@@ -851,3 +851,64 @@ def _website_show_quick_add(self):
...
+        :param dict kwargs: Locally unused data passed to `super`.
+        :rtype: float
+        :return: The specified product's display price.
+        """
+        price = super()._get_configurator_display_price(
+            product_or_template, quantity, date, currency, pricelist, **kwargs
+        )
+
+        website = ir_http.get_request_website()
+        if website:
```
Comment (Feyensv) [inline addons/website_sale/models/product_template.py:876]:
```suggestion
        if website:= ir_http.get_request_website():
```

**Review by Feyensv — APPROVED** [review]
@robodoo delegate+

**Comment by ltinel** [comment]
@robodoo r+

**Pull request closed without merging**